packages/ember-glimmer/tests/integration/mount-test.js

```diff
@@ -188,6 +188,56 @@ moduleFor('{{mount}} test', class extends ApplicationTest {
     });
   }
 
+  ['@test it only unbinds the engines event listeners during teardown']() {
+    this.router.map(function() {
+      this.route('engine-teardown');
+      this.route('engine-teardown-engine');
+    });
+    let controller;
+    this.add('controller:engine-teardown', Controller.extend({
+      totalClicks: 0,
+      actions: {
+        incrementTotalClicks() {
+          this.incrementProperty("totalClicks");
+        }
+      }
+    }));
+    this.addTemplate('engine-teardown', 'Clicks: {{totalClicks}} <button id="increment-engine-total-clicks" onclick={{action "incrementTotalClicks"}}>Increment</button>');
+    this.addTemplate('engine-teardown-engine', '{{mount "foo"}}');
+
+    this.add('engine:foo', Engine.extend({
+      router: null,
+      init() {
+        this._super(...arguments);
+        this.register('template:application', compile('<h2>Foo Engine: {{tagless-component}}</h2>', { moduleName: 'application' }));
+        this.register('component:tagless-component', Component.extend({
+          tagName: ""
+        }));
+        this.register('template:components/tagless-component', compile('Tagless Component', { moduleName: 'components/tagless-component' }));
+      }
+    }));
+
+    return this.visit('/engine-teardown').then(() => {
+      this.assertComponentElement(this.firstChild, { content: 'Clicks: 0 <button id="increment-engine-total-clicks">Increment</button>' });
+
+      this.runTask(() => this.$("#increment-engine-total-clicks").click());
+
+      this.assertComponentElement(this.firstChild, { content: 'Clicks: 1 <button id="increment-engine-total-clicks">Increment</button>' });
+
+      return this.visit('/engine-teardown-engine').then(() => {
```

Comment:
Can you swap this to using links that are clicked (instead of this.visit which I believe creates a new application instance)? Might help repro the original issue without the fix...